### ISSUE TITLE
--save-polygons=geojson only creates .json file not .geojson

### ISSUE THREAD

**Issue by rgill10** [kharchenkolab/Baysor#129]
Hi all! I'm trying to implement baysor with prior segmentation input. I see a --save-polygons feature was added to output a .geojson file in addition to the .parquet output. However, when I run baysor the output is a .json file that is not formated properly as an input for geojson applications. Could someone please let me know if there is anything Im missing? Here is what I am ran

# Step 2: Run Baysor
export JULIA_NUM_THREADS=32
singularity run baysor_v0.6.2.sif run \
--no-ncv-estimation \
--save-polygons=geojson \
-p \
-o baysor_segmentation.csv \
-c input_config3.toml \
input_transcripts.csv :cell_id

Thanks!

**Comment by VPetukhov** [comment]
Hi @rgill10 , the way you run Baysor is correct. Could you please share the error? I tested the formatting in Python geojson package, as well as javascript specification match. But maybe the software you use has stricter requirements.

**Comment by rgill10** [comment]
Hi @VPetukhov, thank you for the reply! I am trying to integrate baysor outputs into Vizgen's VPT pipeline for merfish. I found a workaround script posted by @pakiessling called baysor2vpt.py which converts the current baysor .json output to a VPT compatible .geojson. It would be great if this could be integrated into the baysor pipeline for other users

**Comment by VPetukhov** [comment]
Alright, I'll look into it! [The link to the script](https://github.com/pakiessling/baysor2vpt/blob/main/baysor2vizgen.py) for convenience.

**Comment by GBeattie** [comment]
Just adding as it seems the issue isn't an isolated one, having the same issue, --save-polygons=geojson flag added but output is .json.

**Comment by VPetukhov** [comment]
@GBeattie , could you please clarify your problem? Geojson is `.json`.

**Comment by GBeattie** [comment]
Hey @VPetukhov,

Sure, I'll admit my familiarity with .json/.geojson is minimal, but here is what I can see:

For a different sample I tried the proseg pipeline, which produced a .geojson with the structure seen below, and the file was readable using st_read from which the polygons could be visualised using ggplot;

```
head cell-polygons.geojson
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "properties": {
        "cell": 0
      },
      "geometry": {
        "type": "MultiPolygon",
```

The baysor output looks to me to be (from a little reading) in the .json format, as it lacks the expected header elements and structure as seen in the proseg .gejson output above (it was also not readable by st_read);

```
head -c 300 segmentation_polygons.json
{"geometries":[{"coordinates":[[[84351.3170878093,14882.7264706294],[84344.5539474488,14885.1662874222],[84346.2467193604,14886.2471183141],[84351.7223993937,14888.2170518239],[84359.2802683513,14893.2566245397],[84362.276395162,14891.916513443],[84364.5890553792,14894.0066496531],[84366.7268753052,%
```

For reference, here is my command for the baysor run;

`JULIA_NUM_THREADS=10 /home/regmgbe/Programmes/bin/baysor/bin/baysor run -x x_global_px -y y_global_px -g target -o ./run_out_mini -m 1 -s 10 --save-polygons=geojson --count-matrix-format tsv /home/regmgbe/Scratch/Projects/FastReseg/flatFiles_brain/L1_SU500/perFOV_txFile/fov_100_tx_data.csv`

Thanks for taking a look, apologies if it's a simple lack of understanding on my end!

**Comment by rgill10** [comment]
I noticed the same exact formatting differences between .json output from baysor and other .geojson files that I've used previously!

**Comment by VPetukhov** [comment]
Thank you for the details! I noticed that [10x docs say](https://www.10xgenomics.com/support/software/xenium-ranger/1.7/analysis/inputs/XR-input-overview#compat-files) they're compatible with Baysor. But anyway, I'll change the format to "FeatureCollection", as provided by 10x example.

@GBeattie , there is still a small difference to your example though, as they store cell id as `id` in the `Feature` object, but you have it as `properties/cell`.

I also wonder, what would be the best option to store 3D polygons. Now they're stored as the Dict with `GeometryCollection` per `z` slice. But it isn't exactly compatible with GeoJSON. So far, my best idea is to save two separate files:
1. `polygons_2d.geojson`, which would have molecules pulled across all z slices and output a single `FeatureCollection`
2. There are 2 alternatives here:
    a. `polygons.json`, where all polygons are stored together in a Dict indexed by z as they're now, just using `FeatureCollection` instead of a `GeometryCollection`.
    b. Alternatively, I can pull all polygons into a single `FeatureCollection`, storing z-id in `properties`. It would be compatible with GeoJSON, but wouldn't be as simple to work with from code, as one would need to loop through the whole collection to subset a given z-slice instead of a simple index.

Any suggestions?

**Comment by GBeattie** [comment]
Yes the proseg output is a bit different from what 10X expects. 

I'm afraid I have limited knowledge of the downstream application of the .geojson. My needs were simply to visualise the 2D polygons to visually assess the segmentation, for which the suggested polygons_2d.geojson output would be highly useful, sorry I couldn't add more!

**Comment by VPetukhov** [comment]
This is resolved in v0.7.0. See the [changelog](https://github.com/kharchenkolab/Baysor/blob/master/CHANGELOG.md).
